scarlet/initialization.py

```diff
@@ -48,106 +48,138 @@ def get_best_fit_seds(morphs, images):
+            # best fit solution for the model amplitude of the center pixel
+            # to yield to PSF center: (spectrum * psf_center) / psf_center**2
+            # or shorter:
+            spectrum /= psf_center
+
+        spectra.append(spectrum)
 
-    if np.any(sed <= 0):
+    spectrum = np.concatenate(spectra).reshape(-1)
+
+    if np.any(spectrum <= 0):
         # If the flux in all channels is  <=0,
         # the new sed will be filled with NaN values,
         # which will cause the code to crash later
-        msg = "Zero or negative SED {} at y={}, x={}".format(sed, *sky_coord)
-        if np.all(sed <= 0):
+        msg = "Zero or negative spectrum {} at y={}, x={}".format(spectrum, *sky_coord)
+        if np.all(spectrum <= 0):
             logger.warning(msg)
         else:
             logger.info(msg)
 
-    return sed
+    return spectrum
+
 
+def get_psf_spectrum(sky_coord, observations):
+    """Get spectrum for a point source at `sky_coord` in `observation`
 
-def get_psf_sed(sky_coord, observations, model_frame):
-    """Get SED for a point source at `sky_coord` in `observation`
+    Equivalent to point source photometry for isolated sources. For extended source,
+    this will underestimate the actual source flux in every channel. In case of crowding,
+    the resulting photometry is likely contaminated by neighbors.
 
-    Identical to `get_pixel_sed`, but corrects for the different
-    peak values of the observed seds to approximately correct for PSF
-    width variations between channels.
+    Yields the spectrum of a PSF-homogenized source of flux 1 in every channel,
+    concatenated for all observations.
 
     Parameters
     ----------
     sky_coord: tuple
         Position in the observation
     observations: instance or list of `~scarlet.Observation`
         Observation to extract SED from.
-    model_frame: `~scarlet.Frame`
-        Frame of the model
 
     Returns
     -------
-    SED: `~numpy.array`
+    spectrum: `~numpy.array`
     """
 
+    # assert normalization in ["max", "sum"]
+
     if not hasattr(observations, "__iter__"):
         observations = (observations,)
 
-    seds = []
+    spectra = []
     for obs in observations:
-        sed = get_pixel_sed(sky_coord, obs)
 
-        if type(obs) is LowResObservation:
-            normalization = "sum"
-        else:
-            normalization = "max"
+        pixel = obs.frame.get_pixel(sky_coord)
+        index = np.round(pixel).astype(np.int)
+
+        psf = obs.frame.psf.get_model()
+        bbox = obs.frame.psf.bbox + (0, *index)
+        img = bbox.extract_from(obs.images)
+
+        # img now 0 outside of observation, psf is not:
+        # restrict both to observed pixels to avoid truncation effects
+        mask = img[0] > 0
+        psf = psf[:, mask]  # flattens array in last two axes
+        img = img[:, mask]
+
+        spectrum = (img * psf).sum(axis=1) / (psf * psf).sum(axis=1)
```

Comment:
I'm not sure where that is from. This formula would be correct if two of the psfs here were replaced by the inverse Fourier transform of the inverse of the Fourier transform of the psf, which does not exist since our psfs are band-limited. What gives you the right to use the psf instead?

Reply:
Couldn't we use the inverse of the psf in Fourier with reconditioning?

Reply:
the formula is the amplitude of `img` when projected onto `psf`. In other words, it's the factor you have to multiply `psf` WITH to get `img` (if `img` had the same shape as `psf`).

Reply:
leaving a comment in code now

Reply:
Which is ok for compact source, but a big assumption for extended ones.